src/components/selectorComponents/Checkbox.tsx

```diff
@@ -0,0 +1,27 @@
+import React, { useState } from "react";
+import styles from "./Checkbox.module.css";
+
+interface CheckboxProps {
+  initialState?: boolean;
+  onStateChange?: (newState: boolean) => void;
+}
+
+const Checkbox = (props: CheckboxProps) => {
+  const [state, setState] = useState(true);
+
+  const clickHandler = () => {
+    const newState = !state;
+    setState(newState);
+
+    if (props.onStateChange) {
+      props.onStateChange(newState);
+    }
+  };
+  return (
+    <div className={styles.checkboxWrapper} onClick={clickHandler}>
+      {state && <div className={styles.innerSquare} />}
```

Comment:
**nitpick:** I strongly suggest to put `input` of type `checkbox` with `hidden` attribute, so you keep with semantics. Currently it is not clear for example for screen readers that this component is Checkbox

Reply:
I'll apply this in the next PR, because I made changes there to the Checkbox element.